### PR TITLE
feat: project change request page frontend

### PR DESCRIPTION
**Pull request by Tymek**
## About the changes
- Refactor project menu - merge "Access" and "Environments" with newly added "Change request configuration" into "Project settings" tab.
- Add Change request config page with table
- Add Change request dialog

Closes https://linear.app/unleash/issue/1-344/frontend-project-options
Relates to [roadmap](https://github.com/orgs/Unleash/projects/10) item: #2251

### CODE REVIEW

File: frontend/src/component/project/Project/ProjectSettings/ChangeRequestConfiguration/ChangeRequestConfiguration.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        environment: '',
+    });
+    const projectId = useRequiredPathParam('projectId');
+    const data = [
+        {
+            environment: 'dev',
+            type: 'test',
+            isEnabled: false,
+        },
+    ] as any[]; // FIXME: type
```
Comment (kwasniew) [inline frontend/src/component/project/Project/ProjectSettings/ChangeRequestConfiguration/ChangeRequestConfiguration.tsx:36]: is it not decided yet?

File: frontend/src/component/project/Project/ProjectSettings/ChangeRequestConfiguration/ChangeRequestConfiguration.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+    const onClick = (environment: string) => () => {
```
Comment (kwasniew) [inline frontend/src/component/project/Project/ProjectSettings/ChangeRequestConfiguration/ChangeRequestConfiguration.tsx:38]: rename to enableEnvironment?

File: frontend/src/component/project/Project/ProjectSettings/ChangeRequestConfiguration/ChangeRequestConfiguration.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+            >
+                <Typography>
+                    You are about to enable “Change request”
+                    {dialogState.environment
+                        ? ` for ${dialogState.environment}`
+                        : ''}
+                    .
+                </Typography>
+                <Typography>
+                    When enabling change request for an environment, you need to
```
Comment (kwasniew) [inline frontend/src/component/project/Project/ProjectSettings/ChangeRequestConfiguration/ChangeRequestConfiguration.tsx:150]: in the design there was a picture with the change request flow. are we deliberately skipping it for now?

**Review by kwasniew — APPROVED** [review]
Good one

File: frontend/src/component/project/Project/ProjectSettings/ChangeRequestConfiguration/ChangeRequestConfiguration.tsx (hunk 3 above)
Comment (Tymek) [reply frontend/src/component/project/Project/ProjectSettings/ChangeRequestConfiguration/ChangeRequestConfiguration.tsx:150]: Yes. Designs changed recently. I'll update it in next PR.

File: frontend/src/component/project/Project/ProjectSettings/ProjectSettings.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        { id: 'change-requests', label: 'Change request configuration' },
+    ];
+
+    const onChange = (tab: ITab) => {
+        navigate(tab.id);
+    };
+
+    return (
+        <VerticalTabs
+            tabs={tabs}
```
[the remark this answers is not in the document]
Comment (Tymek) [reply frontend/src/component/project/Project/ProjectSettings/ProjectSettings.tsx:29]: I think this was made to match similar MUI component

File: frontend/src/component/project/Project/ProjectSettings/ChangeRequestConfiguration/ChangeRequestConfiguration.tsx (hunk 1 above)
Comment (Tymek) [reply frontend/src/component/project/Project/ProjectSettings/ChangeRequestConfiguration/ChangeRequestConfiguration.tsx:36]: Will be done when integrating with API

File: frontend/src/component/project/Project/ProjectSettings/ChangeRequestConfiguration/ChangeRequestConfiguration.tsx [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+            },
+            useGlobalFilter
+        );
+
+    return (
+        <PageContent
+            header={<PageHeader titleElement="Change request configuration" />}
+            // isLoading={loading}
+        >
+            <Alert severity="info" sx={{ mb: 3 }}>
```
[the remark this answers is not in the document]
Comment (Tymek) [reply frontend/src/component/project/Project/ProjectSettings/ChangeRequestConfiguration/ChangeRequestConfiguration.tsx:104]: I think it's only available in `sx` prop, and common in the codebase. It's a shorthand for `marginBottom: theme => theme.spacing(3)`

https://mui.com/system/getting-started/the-sx-prop/#spacing

**Pull request merged**